### PR TITLE
gnupg-pkcs11-scd: init at 0.9.2

### PR DESCRIPTION
**Pull request by philandstuff**
###### Motivation for this change

This adds gnupg-pkcs11-scd, a smart card daemon for GnuPG that supports
PKCS#11 smartcards (such as the Yubikey PIV module).

You can use it by adding something like this to your
~/.gnupg/gpg-agent.conf:

    scdaemon-program /home/<user>/.nix-profile/bin/gnupg-pkcs11-scd

You will also need to install `opensc` and have a
~/.gnupg/gnupg-pkcs11-scd.conf with something like the following:

    providers opensc

    provider-opensc-library /home/philandstuff/.nix-profile/lib/pkcs11/opensc-pkcs11.so

Then `gpg` smartcard operations will access your PKCS#11-capable
smartcard.

###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

###### Notify maintainers

cc @

### CODE REVIEW

**Comment by grahamc** [comment]
How does this interact with https://nixos.org/nixos/options.html#services.pcscd.enable?

**Comment by philandstuff** [comment]
@grahamc good question.

Although any given user can only use one of pcscd or gnupg-pkcs11-scd, it doesn't hurt to have both running - ie they don't conflict.  This is because access is controlled via the `scdaemon-program` option of `gpg-agent.conf` - they're not fighting to listen on the same socket or anything.

So it's possible (and indeed I have verified) to use gnupg-pkcs11-scd even if `services.pcscd.enable` is `true`.  But if you're using this, you probably want it to be `false` because you probably won't be using pcscd.

**Comment by philandstuff** [comment]
By the way, I don't like how much manual config is still required to get this working - see the original post / commit message for what I mean.  I'm open to suggestions about how this could be improved.

**Comment by lschuermann** [comment]
I've seen this just by coincidence, as I've packaged the `gnupg-pkcs11-scd` myself a few weeks back and wanted to create the PR at NixCon. :)
The good thing: I came up with pretty much an identical package definition, and can confirm that it works!
I'm working on TPM support currently, so I'll just test this package with my `tpm2-pkcs11` in a few hours.

**Review by lschuermann — APPROVED** [review]

**Comment by philandstuff** [comment]
@lschuermann would you like to add yourself as a maintainer then? :)

**Comment by lschuermann** [comment]
> @lschuermann would you like to add yourself as a maintainer then? :)

As I actually plan on using this for an extended period of time, this is fine with me. That way, we can just shout at each other if something breaks. ;)

File: pkgs/tools/security/gnupg-pkcs11-scd/default.nix [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+
+  meta = with stdenv.lib; {
+    description = "A smart-card daemon to enable the use of PKCS#11 tokens with GnuPG";
+    longDescription = ''
+    gnupg-pkcs11 is a project to implement a BSD-licensed smart-card
+    daemon to enable the use of PKCS#11 tokens with GnuPG.
+    '';
+    homepage = http://gnupg-pkcs11.sourceforge.net/;
+    license = licenses.bsd3;
+    maintainers = with maintainers; [ lschuermann philandstuff ];
```
Comment (lschuermann) [inline pkgs/tools/security/gnupg-pkcs11-scd/default.nix]: Looks good!

**Review by lschuermann — APPROVED** [review]
I am yet to really try this with TPM, but the library seems to be fine so works for me.

**Pull request merged**